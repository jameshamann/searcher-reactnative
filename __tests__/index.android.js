import 'react-native';
import React from 'react';
import Index from '../index.android.js';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <Index />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

//I don't even know why the hell I kept this in.

it('displays searcher on the page', () =>{
  const tree = renderer.create(
    <Index />
  ).toJSON();
  expect(tree.children[0].children).toEqual(["Searcher"])
});


it('displays movie data after submitting a title', () => {
  const url = "https://www.omdbapi.com/?tomatoes=true&t=Primer"
  const mockMovieResponse = {Title: "Primer", Year: "2004"}
  fetch = jest.fn((url, options) => new Promise((resolve, reject) => {
    resolve( { status: 201, json: () => (mockMovieResponse) })
  }));
  expect(mockMovieResponse.Title).toEqual("Primer")
  expect(mockMovieResponse.Year).toEqual("2004")

});
