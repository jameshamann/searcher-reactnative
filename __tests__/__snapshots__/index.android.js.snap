exports[`test renders correctly 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F5FCFF",
          "flex": 1,
          "justifyContent": "center",
          "padding": 5,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "black",
            "fontFamily": "sans-serif-thin",
            "fontSize": 40,
            "margin": 10,
            "textAlign": "center",
          }
        }>
        Searcher
      </Text>
      <View
        style={
          Object {
            "padding": 5,
          }
        }>
        <Image
          source={1}
          style={
            Object {
              "height": 50,
              "width": 50,
            }
          } />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F5FCFF",
            "flex": 1,
            "justifyContent": "center",
            "padding": 5,
          }
        }>
        <TextInput
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          style={
            Object {
              "borderColor": "grey",
              "borderWidth": 1,
              "height": 40,
              "padding": 10,
              "width": 200,
            }
          }
          underlineColorAndroid="transparent"
          value="" />
      </View>
      <View>
        <Image
          source={
            Object {
              "uri": undefined,
            }
          }
          style={
            Object {
              "height": 222,
              "padding": 5,
              "width": 150,
            }
          } />
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "black",
            "fontFamily": "sans-serif-thin",
            "marginBottom": 5,
            "padding": 5,
            "textAlign": "center",
          }
        } />
    </View>
  </View>
</RCTScrollView>
`;
